src/hooks/users.ts

```diff
@@ -69,6 +70,19 @@ export const useWorldUsersById = () => {
   );
 };
 
+// @debt typing, this uses Partial<Record<K,T>> to work around the bug where Record implies that a User will exist for literally any given string, which is untrue
+// @debt refactor all usages of Record<string, User> to Partial<Record<string, User>> (then we can remove this workaround)
+export const useWorldUsersByIdWorkaround = () => {
+  const {
+    worldUsersById: _worldUsersById,
+    isWorldUsersLoaded,
+  } = useWorldUsersById();
+
+  const worldUsersById: Partial<Record<string, User>> = _worldUsersById;
+
+  return { worldUsersById, isWorldUsersLoaded };
+};
```

Comment:
@mike-lvov FYI. Temporary 'centralised' workaround until we refactor all the related code to work well with potentially undefined results.